blue-hat/cbeer/blue_hat.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+  "flag"
+  "fmt"
+  "os"
+  "image"
+  "image/png"
+  "image/color"
+)
+
+func main() {
+  infilePtr := flag.String("infile", "red_hat.png", "Input file name")
+  outfilePtr := flag.String("outfile", "blue_hat.png", "Output file name")
+
+  flag.Parse()
+
+  fmt.Println("input file name:  ", *infilePtr)
+  fmt.Println("output file name: ", *outfilePtr)
+
+  infile, err := os.Open(*infilePtr)
+  if err != nil {
+    fmt.Fprintf(os.Stderr, "Failed to open input file: %v\n", err)
+    os.Exit(1)
+  }
+  defer infile.Close()
+
+  redHatImage, _, err := image.Decode(infile)
+  if err != nil {
+    fmt.Fprintf(os.Stderr, "Failed to decode input file: %v\n", err)
+    os.Exit(1)
+  }
+
+  blueHatImage := image.NewRGBA(redHatImage.Bounds())
+  for y := redHatImage.Bounds().Min.Y; y < redHatImage.Bounds().Max.Y; y++ {
+    for x := redHatImage.Bounds().Min.X; x < redHatImage.Bounds().Max.X; x++ {
+      pixelColor := redHatImage.At(x, y)
+      r, g, b, a := pixelColor.RGBA()
+      if r > (g + b) {
```

Comment:
Why adding `g + b`, comparing red to blue didn't work properly?

Reply:
Actually no need to compare. For this image switching red and blue directly is good enough.

Reply:
The conditional `r > (g + b)` was the suggestion in the chat, I didn't really think about it too hard. I didn't want to just swap everything, since I was looking for a generic answer.